terminus-ui/chip/src/chip-collection.component.ts

```diff
@@ -0,0 +1,588 @@
+    this._userTabIndex = value;
+    this._tabIndex = value;
+  }
+  public get tabIndex(): number {
+    return this._tabIndex;
+  }
+
+  /**
+   * Orientation of the chip collection.
+   */
+  @Input('aria-orientation')
+  public ariaOrientation: 'horizontal' | 'vertical' = 'horizontal';
+
+  /** Event emitted when the chip collection value has been changed by the user. */
+  @Output()
+  public readonly collectionChange = new EventEmitter<TsChipCollectionChange>();
+
+  /**
+   * Emitted when a chip is to be removed.
+   */
+  @Output()
+  public readonly removed = new EventEmitter<TsChipEvent>();
+
+  /**
+   * Emitted when tab pressed with chip focused
+   */
+  @Output()
+  public readonly tabUpdateFocus = new EventEmitter();
+
+  constructor(protected elementRef: ElementRef<HTMLElement>,
+              private changeDetectorRef: ChangeDetectorRef,
+              public zone: NgZone,
+              @Optional() private dir: Directionality,) {
+    zone.runOutsideAngular(() => {
+      this._role = this.empty ? null : 'listbox';
+    });
+  }
+
+  public ngOnInit() {
+    this.selectionModel = new SelectionModel<TsChipComponent>(this.allowMultipleSelections, undefined, false);
+  }
+
+  public ngAfterViewInit() {
+    this.keyManager = new FocusKeyManager<TsChipComponent>(this.chips)
+      .withWrap()
+      .withVerticalOrientation()
+      .withHorizontalOrientation(this.dir ? this.dir.value : 'ltr');
+
+    this.keyManager.tabOut.pipe(untilComponentDestroyed(this)).subscribe(() => {
+      this.tabUpdateFocus.emit();
+    });
+
+    // When the collection changes, re-subscribe
+    this.chips.changes.pipe(startWith(null), untilComponentDestroyed(this)).subscribe(() => {
+      if (this.isDisabled || this.isReadonly) {
+        // Since this happens after the content has been
+        // checked, we need to defer it to the next tick.
+        Promise.resolve().then(() => {
+          this.syncChipsState();
+        });
+      }
+
+      this.resetChips();
+
+      // Check to see if we need to update our tab index
+      Promise.resolve().then(() => {
+        this.updateTabIndex();
+      });
+
+      // Check to see if we have a destroyed chip and need to refocus
+      this.updateFocusForDestroyedChips();
+      this.propagateChanges();
+
+    });
+  }
+
+  public ngOnDestroy() { }
+
+  /**
+   * When blurred, mark the field as touched when focus moved outside the chip collection.
+   */
+  public blur() {
+    if (!this.focused) {
+      this.keyManager.setActiveItem(-1);
+    }
+  }
+
+  public remove(event) {
+    this.removed.emit(event);
+  }
+
+  /**
+   * Focuses the first non-disabled chip in this chip collection, or the associated input when there
+   * are no eligible chips.
+   */
+  public focus(): void {
+    if (this.isDisabled) {
+      return;
+    }
+
+    if (this.chips.length > 0) {
+      this.keyManager.setFirstItemActive();
+    }
+  }
+
+  /**
+   * Utility to for whether input field is empty
+   * @param element
```

Comment:
Missing line break before `@param` and has extra line break between comment and function